packages/e2e-tests/specs/editor/various/rtl.test.js

```diff
@@ -104,6 +104,9 @@ describe( 'RTL', () => {
 		await page.evaluate( () => ( document.dir = 'rtl' ) );
 		await page.keyboard.press( 'Enter' );
 
+		// Wait for rich text editor to load.
+		await page.waitForSelector( '.block-editor-rich-text__editable' );
```

Comment:
This is also used here:

https://github.com/WordPress/gutenberg/blob/cb10e2d6e249000858da4eb3fc07cc44c832eddb/packages/e2e-tests/specs/editor/various/rich-text.test.js#L256-L262

Reply:
The reason why this is sometimes needed: caret position is not possible to be set internally upon focus. RichText only knows this information after a small delay. That's ok for typing since that's done by the browser, but it is needed for applying formats.